### ISSUE TITLE
Code suggestion suggests a variable called tokens which doesn't exist

### ISSUE THREAD

**Issue by ALiwoto** [PowerShell/vscode-powershell#4852]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [_open and closed_ issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/main/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [X] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/main/SECURITY.md).

### Summary

Sometimes when I try to type `$true` in the code editor, PowerShell extension will suggest me a variable called `$tokens`. Is this variable perhaps resulted from code parsing?

### PowerShell Version

```console
PS> $PSVersionTable; $HOST

Name                           Value
----                           -----
PSVersion                      7.3.7
PSEdition                      Core
GitCommitId                    7.3.7
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

Name             : Visual Studio Code Host
Version          : 2023.8.0
InstanceId       : 4d9cd2d0-b385-4319-9e40-e978be5a7e98
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : en-US
CurrentUICulture : en-US
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace
```

### Visual Studio Code Version

```console
PS> code --version
1.84.2
1a5daa3a0231a0fbba4f14db7ec463cf99d7768e
x64
```

### Extension Version

```console
PS> code --list-extensions --show-versions | Select-String powershell

ms-vscode.powershell@2023.8.0
```

### Steps to Reproduce

1. Try to write something that will invoke the code-suggestion functionality (e.g. `Parameter(Mandatory = $t`).
2. See the parameters list. Sometimes (most of the times) a variable will appear called `$tokens`.

### Visuals

![image](https://github.com/PowerShell/vscode-powershell/assets/70187458/5d0fff80-0e6b-4f14-894b-bbfd27139660)

![image](https://github.com/PowerShell/vscode-powershell/assets/70187458/c4ec9845-ebc6-4aa9-b406-443bdff172c7)

**Comment by andyleejordan** [comment]
That looks like it's coming from the expand alias handler:

[$tokens = [System.Management.Automation.PsParser]::Tokenize($targetScript, $errors).Where({$_.type -eq 'command'}) |](https://github.com/PowerShell/PowerShellEditorServices/blob/37f0648ca28358c1fd2d0688801658948f69090d/src/PowerShellEditorServices/Services/PowerShell/Handlers/ExpandAliasHandler.cs#L41-L42)

**Comment by SydneyhSmith** [comment]
Thanks for the issue @ALiwoto we believe this may be fixed in PowerShell 7.4, let us know if this issue still reproduces for you with PS 7.4
